acme_diags/driver/tc_analysis_driver.py

```diff
@@ -0,0 +1,541 @@
+from __future__ import print_function
+
+import collections
+import os
+from datetime import datetime, timedelta
+from typing import TYPE_CHECKING, Any, Dict, List, Tuple
+
+import cdms2
+import numpy as np
+from netCDF4 import Dataset as netcdffile
+
+import acme_diags
+from acme_diags.plot.cartopy import tc_analysis_plot
+
+if TYPE_CHECKING:
+    from numpy.ma.core import MaskedArray
+
+    from acme_diags.parameter.core_parameter import CoreParameter
+
+
+# Years include 1979–2018 according to Balaguru et al. 2020
+OBS_START_YR = 1979
+OBS_END_YR = 2018
+OBS_YEARS = np.arange(OBS_START_YR, OBS_END_YR + 1)
+
+# (basin name, E bound, W bound, S bound, N bound, observed hurricane number per year)
+BasinInfo = Tuple[str, float, float, float, float, float]
+BASIN_DICT: Dict[str, BasinInfo] = {
+    "NA": ("North Atlantic", 270, 360, 0, 45, 8.6),
+    "WP": ("Northwest Pacific", 100, 180, 0, 45, 26.7),
+    "EP": ("Eastern Pacific", 180, 270, 0, 45, 18.1),
+    "NI": ("North Indian", 30, 100, 0, 45, 4.6),
+    "SI": ("South Indian", 20, 135, -45, 0, 15.4),
+    "SP": ("South Pacific", 135, 270, -45, 0, 10.4),
+}
+
+
+def run_diag(parameter: "CoreParameter") -> "CoreParameter":
```

Comment:
`run_diag` moved to the top.

Using the "Stepdown" method of ordering functions in a hierarchy.